middleman-core/lib/middleman-core/core_extensions/i18n.rb

```diff
@@ -256,10 +273,20 @@ def parse_locale_extension(path)
     [locale, path, basename]
   end
 
-  LocalizedPageDescriptor = Struct.new(:path, :source_path, :locale) do
+  # Turn path into a page_id for the lookup map
+  # eg. "foo/bar/baz.foo.html" => "foo.bar.baz.foo"
```

Comment:
Why is the actual path string not enough? What does changing this into all dots get us? Is it intended to be an actual structure from a locale file?